### PR TITLE
Add UT for TransportLoadModelAction

### PR DESCRIPTION
**Pull request by zane-neo**
Signed-off-by: Zan Niu <zaniu@amazon.com>

### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ylwu-amzn** [comment]
hi, @zane-neo Thanks for helping improve code coverage. Seems the code style check failed in Github CI. Suggest run `./gradlew spotlessApply` and publish a new commit.

File: plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java [hunk 1]
```diff
@@ -106,130 +107,121 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<LoadMo
...
-                    eligibleNodes.add(node);
-                }
-            }
-            if (nodeIds.size() == 0) {
-                listener.onFailure(new MLResourceNotFoundException("no eligible node found"));
-                return;
             }
+        } else {
+            nodeIds.addAll(allEligibleNodeIds);
+            eligibleNodes.addAll(Arrays.asList(allEligibleNodes));
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java:129]: Thanks for tuning this part. Not sure if this code will perform better than the original one. I guess they are almost the same. But prefer to learn any result if you have done some experiments. Here the allEligibleNodes will be a relatively small number like 100s.

File: plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java [hunk 2]
```diff
@@ -106,130 +107,121 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<LoadMo
...
-                        .createTime(Instant.now())
-                        .lastUpdateTime(Instant.now())
-                        .state(MLTaskState.CREATED)
-                        .workerNode(workerNodes)
-                        .build();
-                    mlTaskManager.createMLTask(mlTask, ActionListener.wrap(response -> {
-                        String taskId = response.getId();
-                        mlTask.setTaskId(taskId);
-
-                        try {
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java:163]: I think we should keep this try block, this was to catch exception of different parts, will print different exception message and also remove task `mlTaskManager.remove(taskId);`

Comment (zane-neo) [reply plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java:163]: Added this try catch back.

File: plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java (hunk 1 above)
Comment (zane-neo) [reply plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java:129]: Checked the code, `Arrays.asList` construct a new object without any iteration, `addAll` method has similar logic with `add` so there's no extra time consuming. Also did a test with 120 records, this approach has faster completion time.

File: plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java [hunk 3]
```diff
@@ -106,134 +107,138 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<LoadMo
...
+                    .workerNode(workerNodes)
+                    .build();
+                mlTaskManager.createMLTask(mlTask, ActionListener.wrap(response -> {
+                    String taskId = response.getId();
+                    mlTask.setTaskId(taskId);
+                    try {
+                        mlTaskManager.add(mlTask, nodeIds);
+                        listener.onResponse(new LoadModelResponse(taskId, MLTaskState.CREATED.name()));
+                        threadPool
+                            .executor(TASK_THREAD_POOL)
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/action/load/TransportLoadModelAction.java:164]: Tuned thread pool, so there will be conflict for this part

**Review by ylwu-amzn — DISMISSED** [review]

**Review by Zhangxunmt — APPROVED** [review]
This TransportLoadModelAction is still in the jacocoExclusions list. Did you check how much percentage for the coverage now in this class?

**Review by ylwu-amzn — APPROVED** [review]

**Review by Zhangxunmt — APPROVED** [review]

**Pull request merged**